src/useClickOutside.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback, useEffect } from 'react';
+
+export function useClickOutside(
+  elRef: React.RefObject<HTMLElement>,
+  callback: () => void,
+) {
+  const handleClickOutside = useCallback(
+    (e: MouseEvent) => {
+      if (!elRef.current) {
+        return;
+      }
+
+      // Check if the clicked target is outside of the referenced element
+      if (!elRef.current.contains(e.target as Node)) {
+        callback();
+      }
+    },
+    [callback, elRef],
+  );
+
+  useEffect(() => {
+    // Add click event listener when component mounts
+    document.addEventListener('click', handleClickOutside, { capture: true });
```

Comment:
if you're using capture, should you stopPropagation to skip bubbling altogether?

Reply:
Do we want to skip bubbling entirely? It seems beneficial in situations where the parent components might need to handle / know about clicks

I like using capture because it doesn't interrupt event flow, just determines when the handler runs.